### PR TITLE
Update CoreFX-in-CoreCLR testing instructions

### PR DESCRIPTION
**Pull request by BruceForstall**
Update to reflect changes in the repo, and fix various formatting issues.

### CODE REVIEW

File: Documentation/building/testing-with-corefx.md [hunk 1]
```diff
@@ -3,134 +3,144 @@ Testing with CoreFX
...
 It may be valuable to use CoreFX tests to validate your changes to CoreCLR or mscorlib.
 
-## Building CoreFX against CoreCLR
+# Building CoreFX against CoreCLR
 **NOTE:** The `BUILDTOOLS_OVERRIDE_RUNTIME` property no longer works.
 
 To run CoreFX tests with an updated System.Private.Corelib.dll, [use these instructions](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits).
 
-To build CoreFX against the updated System.Private.Corelib.dll - we need to update instructions.
+To build CoreFX against the updated System.Private.Corelib.dll - _we need to update instructions_.
```
Comment (jkotas) [inline Documentation/building/testing-with-corefx.md]: The instruction from https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bit on the previous line cover both building the CoreFX product bits and running the tests. We can drop this line.

File: Documentation/building/testing-with-corefx.md [hunk 2]
```diff
@@ -3,134 +3,144 @@ Testing with CoreFX
 
 It may be valuable to use CoreFX tests to validate your changes to CoreCLR or mscorlib.
```
Comment (jkotas) [inline Documentation/building/testing-with-corefx.md]: Nit: mscorlib -> System.Private.Corelib

Comment (jkotas) [reply Documentation/building/testing-with-corefx.md]: Or just say "validate your changes to CoreCLR." and drop mscorlib/CoreLib.

File: Documentation/building/testing-with-corefx.md [hunk 3]
```diff
@@ -3,134 +3,144 @@ Testing with CoreFX
...
+# Building CoreFX against CoreCLR
 **NOTE:** The `BUILDTOOLS_OVERRIDE_RUNTIME` property no longer works.
 
 To run CoreFX tests with an updated System.Private.Corelib.dll, [use these instructions](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits).
 
-To build CoreFX against the updated System.Private.Corelib.dll - we need to update instructions.
+To build CoreFX against the updated System.Private.Corelib.dll - _we need to update instructions_.
 
-**Replace runtime between build.[cmd|sh] and build-tests.[cmd|sh]**
+## Replace runtime between build.[cmd|sh] and build-tests.[cmd|sh]
```
Comment (jkotas) [inline Documentation/building/testing-with-corefx.md]: I would be useful to say something about when one should use these instructions vs. the instructions from https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits

(I always use the CoreFX "testing-with-private-coreclr-bits" for my own workflow.)

File: Documentation/building/testing-with-corefx.md [hunk 4]
```diff
@@ -3,134 +3,144 @@ Testing with CoreFX
...
-Use the following instructions to test a change to the dotnet/coreclr repo using dotnet/corefx tests.  Refer to the [CoreFx Developer Guide](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md) for information about CoreFx build scripts.
+Use the following instructions to test a change to the dotnet/coreclr repo using dotnet/corefx tests. Refer to the [CoreFx Developer Guide](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md) for information about CoreFx build scripts.
 
-1. Build the CoreClr runtime you wish to test under `<coreclr_root>`
-2. Build the CoreFx repo (`build.[cmd|sh]`) under `<corefx_root>`, but don't build tests yet
+1. Build the CoreClr runtime you wish to test under `<coreclr_root>`.
+2. Build the CoreFx repo (`build.[cmd|sh]`) under `<corefx_root>`, but don't build tests yet.
 3. Copy the contents of the CoreCLR binary root you wish to test into the CoreFx runtime folder (`<flavor>` below) created in step #2.  For example:
 
   `copy <coreclr_root>\bin\Product\Windows_NT.<arch>.<build_type>\* <corefx_root>\bin\runtime\<flavor>`
```
Comment (jkotas) [inline Documentation/building/testing-with-corefx.md]: The corefx bin directory is called `artifacts` now.

File: Documentation/building/testing-with-corefx.md [hunk 5]
```diff
@@ -1,118 +1,174 @@
...
 
-## Building CoreFX against CoreCLR
-**NOTE:** The `BUILDTOOLS_OVERRIDE_RUNTIME` property no longer works.
+1. Build the CoreFX product and tests against a build of CoreCLR, or
+2. Use a snapshot of the CoreFX test build with a build of CoreCLR.
 
-To run CoreFX tests with an updated System.Private.Corelib.dll, [use these instructions](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#testing-with-private-coreclr-bits).
+Both mechanisms are exposed to certain types of breaking changes which can cause test failures.
+However, we have a test exclusion mechanism for option #2, which exclusions specified in the
+CoreCLR tree, not the CoreFX tree, which can make it possible to exclude tests that fail for
```
Comment (jkotas) [inline Documentation/building/testing-with-corefx.md]: typo: which -> with ?

**Comment by BruceForstall** [comment]
@jkotas thanks for all the great suggestions.

**Pull request merged**